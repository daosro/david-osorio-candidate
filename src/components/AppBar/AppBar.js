import React from "react";

import { MdPerson, MdSearch } from "react-icons/md";

import useStyles from "./AppBar.style";

const AppBar = () => {
  const classes = useStyles();
  return (
    <header className={classes.root}>
      <MdPerson className={classes.userIcon} />
      <img
        className={classes.logo}
        src={process.env.PUBLIC_URL + "/assets/images/norigin-media-logo.png"}
        alt="NoriginMedia"
      />
      <MdSearch className={classes.searchIcon} />
    </header>
  );
};

export default AppBar;
