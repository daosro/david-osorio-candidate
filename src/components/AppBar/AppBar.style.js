import { createUseStyles } from 'react-jss';

export default createUseStyles({
  root: {
    backgroundColor: '#000000',
    display: 'flex',
    width: '100%',
    height: '80px',
  },
}, { name: 'AppBar' });