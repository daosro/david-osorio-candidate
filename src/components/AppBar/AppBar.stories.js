import React from "react";
import { MemoryRouter } from "react-router";

import AppBar from "./AppBar";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "NoriginMedia/AppBar",
  component: AppBar,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  // Sets the layout parameter component wide.
  argTypes: {
    simple: { control: { type: "boolean" } },
  },
  decorators: [
    (Story) => (
      <MemoryRouter>
        <Story />
      </MemoryRouter>
    ),
  ],
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => <AppBar {...args} />;

export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  simple: false,
};

