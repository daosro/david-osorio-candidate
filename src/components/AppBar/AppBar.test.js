import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import AppBar from "./AppBar";

describe("AppBar", () => {
  it("should render correctly", () => {
    const { container } = render(
      <MemoryRouter>
        <AppBar />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
  it("should contain the logo image", () => {
    const { container } = render(
      <MemoryRouter>
        <AppBar />
      </MemoryRouter>
    );
    expect(container.querySelector("img")).toBeInTheDocument();
  });
});
