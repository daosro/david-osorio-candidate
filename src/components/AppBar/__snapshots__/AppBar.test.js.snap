// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar should render correctly 1`] = `
<div>
  <header
    class="AppBar-root-0-2-1 AppBar-root-d0-0-2-6"
  >
    <svg
      class="AppBar-icon-0-2-2 AppBar-marginLeft-0-2-3"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
    <img
      alt="NoriginMedia"
      class="AppBar-logo-0-2-5"
      src="/assets/images/norigin-media-logo.png"
    />
    <svg
      class="AppBar-icon-0-2-2 AppBar-marginRight-0-2-4"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
  </header>
</div>
`;
